[framework] Remove var dumps from API response (#2536)
shopsys/shopsys: docs/cookbook/dump-variables.md

# Dump variables
Shopsys Framework uses Symfony with its VarDumper. 
In addition to the standard function `dump()`, we implement the function `d()`, which you can pass any number of attributes where each one is dumped.

Dumped variables are primarily available in the profiler. 

Since we do not want to dump variables into the JSON response when developing the frontend API, the dumps are directed to the Symfony dump server. (https://symfony.com/doc/current/components/var_dumper.html#the-dump-server)

You can find the debug settings in the file `config/packages/dev/debug.yaml`:

> Start the server with the `server:dump` command and whenever you call the `d()`, the dumped data will not be displayed in the output but sent to that server, which outputs it to its own console or to an HTML file.

Reviewer: So do I understand it correctly that I have to run the command every time I want the dumps in the server console instead of the output? Or with the new settings, I do not need to do anything and I can always find the dumps in the profiler and they will not mess up the API response anymore out of the box?

Reviewer 2: No, you don't. Dumps are still in profiller. Only mess from API responses are moved into `server:dump` :)

> # Dump variables in framework

Reviewer 3: ```suggestion
# Dump Variables
```

Reviewer 3: We have some rules for writing docs, just like this one, see:
https://docs.shopsys.com/en/latest/contributing/guidelines-for-writing-documentation/

Maybe it would not be bad to read all contributing articles in our docs:
https://docs.shopsys.com/en/latest/contributing/

Shopsys framework uses symfony with its VarDumper. In addition to the standard function `dump()`, we implement the function `d()`, 

> which you can pass any number of attributes where each one is dumped.

Author: ok

Dumped variables are primarily available in the profiler. 

> Since we don't want to dump variables into the json response when developing the frontend api, the dumps are directed to the symfony dump server. (https://symfony.com/doc/current/components/var_dumper.html#the-dump-server)

Reviewer 3: ```suggestion
Since we do not want to dump variables into the JSON response when developing the frontend API, the dumps are directed to the Symfony dump server. (https://symfony.com/doc/current/components/var_dumper.html#the-dump-server)
```

Reviewer 3: It is more formal way to write do not vs don't in English.

https://allthedifferences.com/difference-between-do-not-and-dont/

Author: ok

You can find the debug settings in the file `config/packages/dev/debug.yaml`:
```yaml
dump_destination: tcp://0.0.0.0:9912

> ```

Author: ok

> Start the server with the server:dump command and whenever you call to `d()`, the dumped data won't be displayed in the output but sent to that server, which outputs it to its own console or to an HTML file:

Reviewer 3: ```suggestion
Start the server with the `server:dump` command and whenever you call the `d()`, the dumped data will not be displayed in the output but sent to that server, which outputs it to its own console or to an HTML file.
```

Author: ok
